diff --git a/node_modules/@notum-cz/strapi-plugin-content-versioning/admin/src/components/Versions/index.js b/node_modules/@notum-cz/strapi-plugin-content-versioning/admin/src/components/Versions/index.js
index 95d5291..51ba827 100644
--- a/node_modules/@notum-cz/strapi-plugin-content-versioning/admin/src/components/Versions/index.js
+++ b/node_modules/@notum-cz/strapi-plugin-content-versioning/admin/src/components/Versions/index.js
@@ -84,7 +84,9 @@ const Versions = () => {
         return;
       }
 
-      const selectedVersion = data.find((v) => v.versionNumber === value);
+      const selectedVersion = data.find((v) => {
+        return v.versionNumber == parseInt(value)
+      });
 
       push({
         pathname: `/content-manager/collectionType/${slug}/${selectedVersion.id}`,
