diff --git a/node_modules/strapi-plugin-populate-deep/server/bootstrap.js b/node_modules/strapi-plugin-populate-deep/server/bootstrap.js
index cb357c6..768359b 100644
--- a/node_modules/strapi-plugin-populate-deep/server/bootstrap.js
+++ b/node_modules/strapi-plugin-populate-deep/server/bootstrap.js
@@ -5,7 +5,10 @@ module.exports = ({ strapi }) => {
   // Subscribe to the lifecycles that we are intrested in.
   strapi.db.lifecycles.subscribe((event) => {
     if (event.action === 'beforeFindMany' || event.action === 'beforeFindOne') {
-      const populate = event.params?.populate;
+      let populate = event.params?.populate
+      if (typeof populate === 'string') {
+        populate = [populate]
+      }
       const defaultDepth = strapi.plugin('strapi-plugin-populate-deep')?.config('defaultDepth') || 5
 
       if (populate && populate[0] === 'deep') {
